### PR TITLE
[BasicUI] Add the "follow system" theme

### LINKED ISSUE #1828: [BasicUI] Offer theme following system
## Which UI are you reporting an issue for?

- [x] Basic UI
- [ ] HABPanel
- [ ] HABot
- [ ] CometVisu

## The problem

When the desktop is in “dark mode”, and the browser loads pages in “dark mode” I want to see BasicUI in dark mode, too.

With openHAB 3.4 this is not the case

### PR DESCRIPTION
**Pull request by lolodomo**
Closes #1828

3 themes are now available! bright, dark and system.
Theme previously named "default" has been renamed into "bright".
"default" is still accepted for backward compatibility.

Tested with Fiorefox, Chrome and Edge on Windows 10.
Tested with Chrome on Android.

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by dilyanpalauzov** [comment]
> I tried now the explicit Dark mode. I have to say that for SetPoint items the UP and DOWN arrow are black on dark-grey background and therefore hard to perceive in openHAB 3.4/Basic UI. For Switch item=r3_manual mappings=[ON="Ръчно", OFF="Автоматично"] when the current value is OFF the text “Ръчно” is presented in black on dark--gray background. This has very low contrast and is therefore suboptimal.

Is this also addressed here?

**Comment by lolodomo** [comment]
No, as mentioned in the PR title, this PR adds the new "follow system" theme, meaning an automatic choice between bright and dark themes depending on the system current mode.
This PR makes no change on colors used by the bright and dark themes.

**Comment by dilyanpalauzov** [comment]
Regarding the low contrast for Basic UI in dark mode I filled https://github.com/openhab/openhab-webui/issues/1831 .

My understanding of the web interfaces in openHAB 3.4 so far is:
* For MainUI any unauthenticated user can visit `/about` and setup the “Mode preferences”: Auto (System preference), Light, Dark; and the “Enable Web Audio sink support”.  These settings are per device (per browser, actually)
* The above settings are however irrelevant for BasicUI.  To adjust the latter one has to visit `/settings/services/org.openhab.basicui`.  Then the settings are not per device, but for all BasicUI instances, whereever they are loaded.  Moreover to alter that settings the user has to authenticate towards openHAB.

For me these two-fold settings are contradicting each other.  May I suggest to remove from the URL `/settings/services/org.openhab.basicui` the settings, which are duplicated from `/about` (like color mode and Web Audio sink preferences). Then use on the device where BasicUI is loaded the preferences set at `/about`?  This will simplify things.

**Comment by lolodomo** [comment]
Please create a separate feature request and avoid polluate this PR. I will then comment.

**Comment by lolodomo** [comment]
I have to update this PR now that PR #1834 has been merged.

**Comment by lolodomo** [comment]
PR is again ready for review.

File: bundles/org.openhab.ui.basic/src/main/resources/OH-INF/config/config.xml [hunk 1]
```diff
@@ -10,10 +10,11 @@
 			<label>Theme</label>
 			<description>Defines the UI theme.</description>
 			<options>
-				<option value="default">Default</option>
+				<option value="bright">Bright</option>
 				<option value="dark">Dark</option>
+				<option value="system">Follow system</option>
```
Comment (kaikreuzer) [inline bundles/org.openhab.ui.basic/src/main/resources/OH-INF/config/config.xml:15]:
```suggestion
				<option value="system">Follow System</option>
```

File: bundles/org.openhab.ui.basic/src/main/resources/OH-INF/i18n/basic.properties [hunk 2]
```diff
@@ -18,8 +18,9 @@ ui.config.basic.enableIcons.option.true = Enable
 ui.config.basic.enableIcons.option.false = Disable
 ui.config.basic.theme.label = Theme
 ui.config.basic.theme.description = Defines the UI theme.
-ui.config.basic.theme.option.default = Default
+ui.config.basic.theme.option.bright = Bright
 ui.config.basic.theme.option.dark = Dark
+ui.config.basic.theme.option.system = Follow system
```
Comment (kaikreuzer) [inline bundles/org.openhab.ui.basic/src/main/resources/OH-INF/i18n/basic.properties:23]:
```suggestion
ui.config.basic.theme.option.system = Follow System
```

**Review by kaikreuzer — CHANGES_REQUESTED** [review]
Thanks - just a very minor change request.

File: bundles/org.openhab.ui.basic/src/main/resources/OH-INF/i18n/basic.properties (hunk 2 above)
Comment (lolodomo) [reply bundles/org.openhab.ui.basic/src/main/resources/OH-INF/i18n/basic.properties]: Done

**Review by kaikreuzer — APPROVED** [review]
Thanks!

**Pull request merged**